### PR TITLE
Fix examples that advancing legIndex manually 

### LINKED ISSUE #154: Waypoint-Arrival-Screen uses incorrect API to advance index
https://github.com/mapbox/mapbox-navigation-ios-examples/blob/7fae966a603f85e63b0bb0a9c2557066967383a8/Navigation-Examples/Examples/Waypoint-Arrival-Screen.swift#L53

Advance legIndex by manually changing it. It is not a proper way to change legIndex anymore. https://github.com/mapbox/mapbox-navigation-ios/blob/4d5138008b441168c9cb976d90fd84fe1c44bd10/Sources/MapboxCoreNavigation/RouteController.swift#L130 should be used instead.

### PR DESCRIPTION
**Pull request by S2Ler**
1. advanceLegIndex is the correct API for advancing legIndex
2. I've also added shared schema for examples target. It was remove it seems. 

Fixes #154

### CODE REVIEW

File: Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (azarovalex) [inline Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme:1]: Do we need this file?

**Review by azarovalex — APPROVED** [review]

File: Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme (hunk 1 above)
Comment (S2Ler) [reply Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme:1]: Good question, why do you think we don't need it? I opened the project and there wasn't a scheme to run examples.

Comment (bamx23) [reply Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme:1]: Isn't this repo require open xcworkspace built by CocoaPods?

Comment (S2Ler) [reply Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme:1]: Yeah, I open it but there is no scheme to run examples by default:
Screenshot 2021-11-18 at 19 12 15

Comment (azarovalex) [reply Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme:1]: Strange, but I have this scheme on main branch
image

Comment (S2Ler) [reply Navigation-Examples.xcodeproj/xcshareddata/xcschemes/Navigation-Examples.xcscheme]: Ok, Xcode is crazy. Removing.

**Review by bamx23 — APPROVED** [review]

**Pull request merged**